File: frontend/types/log.ts
export enum SeverityLevel {
  DEBUG = "DEBUG",
  INFO = "INFO",
  WARNING = "WARNING",
  ERROR = "ERROR",
  CRITICAL = "CRITICAL",
}

export interface Log {
  id: number;
  timestamp: string;
  message: string;
  severity: SeverityLevel;
  source: string;
}

export interface LogCreate {
  message: string;
  severity: SeverityLevel;
  source: string;
  timestamp?: string;
}

export interface LogUpdate {
  message?: string;
  severity?: SeverityLevel;
  source?: string;
  timestamp?: string;
}

export interface LogListResponse {
  items: Log[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

export interface LogFiltering {
  severity?: string;
  source?: string;
  count: number;
  date?: string;
}

export interface LogFilteringResponse {
  aggregations: LogFiltering[];
  total_count: number;
}

export interface LogFilters {
  page?: number;
  page_size?: number;
  severity?: SeverityLevel | string;
  source?: string;
  start_date?: string;
  end_date?: string;
  search?: string;
  sort_by?: string;
  sort_order?: 'asc' | 'desc';
}

File: app/schemas/log.py
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional, List
from enum import Enum

class SeverityLevel(str, Enum):
    DEBUG = "DEBUG"
    INFO = "INFO"
    WARNING = "WARNING"
    ERROR = "ERROR"
    CRITICAL = "CRITICAL"

class LogBase(BaseModel):
    message: str = Field(..., min_length=1, max_length=5000)
    severity: SeverityLevel
    source: str = Field(..., min_length=1, max_length=255)

class LogCreate(LogBase):
    timestamp: Optional[datetime] = None

class LogUpdate(BaseModel):
    message: Optional[str] = Field(None, min_length=1, max_length=5000)
    severity: Optional[SeverityLevel] = None
    source: Optional[str] = Field(None, min_length=1, max_length=255)
    timestamp: Optional[datetime] = None

class LogResponse(LogBase):
    id: int
    timestamp: datetime

    class Config:
        from_attributes = True

class LogListResponse(BaseModel):
    items: List[LogResponse]
    total: int
    page: int
    page_size: int
    total_pages: int

class LogFiltering(BaseModel):
    severity: Optional[str] = None
    source: Optional[str] = None
    count: int
    date: Optional[str] = None

class LogFilteringResponse(BaseModel):
    aggregations: List[LogFiltering]
    total_count: int
